### DISCUSSION TITLE
[Question] NFC Features

### DISCUSSION THREAD

**Discussion by cfontana0** [alexandercerutti/passkit-generator#113] [Q&A] [answered]
Hello there 👋
Thanks for your hard work!

I'd like to know if I can use the NFC feature on the Apple Wallet when I add the pass to it.

For instance, can I add an access key to unlock a smart lock?

Thanks!

**Comment by alexandercerutti** [comment, marked as answer]
Hey @cfontana0, thanks for using passkit-generator!

I guess so! I actually don't know how NFC side works. I mean, aside of setting an encryptionKey and a message through [`setNFC` method](https://github.com/alexandercerutti/passkit-generator/wiki/API-Documentation-Reference#setting-nfc-support), the "server side" is under NDA by Apple, and requires you some VAS-compliant (Value Added Services Protocol) terminals (VAS is under NDA).

So I always been limited by this in terms of testing NFC and Personalization. So I don't think I'll be able to help you any further on this topic.

But please, if you are able to do something with NFC and it works, let me know. If I remember correctly, I've set a warning message in the API doc saying what I told you up here.

Let me know

**Comment by cfontana0** [comment]
Ok! Awesome! I'll work on that and let you know!

**Reply by PiyushSaklani** [reply]
Hey @cfontana0, are you able to use NFC? If so, I have a few questions I’d like to ask.

**Reply by alexandercerutti** [reply]
@PiyushSaklani hey, as you might understand this discussion is 2 years old in a few days, so I wouldn't wait for a reply. You can perhaps ask them directly or open an issue and see if others can help you
